app/src/main/java/com/hedvig/app/feature/crossselling/ui/CrossSellingResultScreen.kt

```diff
@@ -0,0 +1,180 @@
+import androidx.compose.foundation.layout.Spacer
+import androidx.compose.foundation.layout.fillMaxSize
+import androidx.compose.foundation.layout.fillMaxWidth
+import androidx.compose.foundation.layout.padding
+import androidx.compose.foundation.layout.size
+import androidx.compose.foundation.layout.width
+import androidx.compose.material.MaterialTheme
+import androidx.compose.material.Text
+import androidx.compose.runtime.Composable
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.res.painterResource
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.compose.ui.tooling.preview.PreviewParameter
+import androidx.compose.ui.tooling.preview.PreviewParameterProvider
+import androidx.compose.ui.unit.dp
+import com.hedvig.app.R
+import com.hedvig.app.ui.compose.HedvigTheme
+import com.hedvig.app.ui.compose.designsystem.LargeContainedButton
+import com.hedvig.app.ui.compose.designsystem.LargeOutlinedButton
+import java.time.LocalDate
+import java.time.format.DateTimeFormatter
+
+// TODO text resources still missing
+@Composable
+fun CrossSellingResultScreen(
+    crossSellingResult: CrossSellingResult
+) {
+    HedvigTheme {
+        Box(
+            modifier = Modifier
+                .fillMaxSize()
+                .padding(horizontal = 16.dp)
+        ) {
+            InformationSection(
+                crossSellingResult = crossSellingResult,
+                modifier = Modifier
+                    .align(Alignment.TopStart)
+                    .padding(top = 96.dp)
+                    .fillMaxWidth(),
+            )
+            ButtonsSection(
+                crossSellingResult = crossSellingResult,
+                modifier = Modifier
+                    .align(Alignment.BottomStart)
+                    .padding(bottom = 16.dp)
+                    .fillMaxWidth(),
+            )
+        }
+    }
+}
+
+@Composable
+private fun InformationSection(
+    crossSellingResult: CrossSellingResult,
+    modifier: Modifier
+) {
+    @DrawableRes
+    val drawableId: Int = when (crossSellingResult) {
+        is CrossSellingResult.Success -> R.drawable.ic_checkmark_in_circle
+        CrossSellingResult.Error -> R.drawable.ic_x_in_circle
+    }
+    val titleText = when (crossSellingResult) {
+        CrossSellingResult.Error -> "Something went wrong."
+        is CrossSellingResult.Success -> {
+            stringResource(
+                R.string.purchase_confirmation_new_insurance_today_app_state_title,
+                crossSellingResult.insuranceType
+            )
+        }
+    }
+    val descriptionText = when (crossSellingResult) {
+        CrossSellingResult.Error -> "Your purchase couldn't be completed.\nContact us in the chat."
+        is CrossSellingResult.Success -> {
+            when {
+                crossSellingResult.startingDate.dayOfMonth <= LocalDate.now().dayOfMonth -> {
```

Comment:
Is this logic correct? Shouldn't it be `crossSellingResult.startingDay == LocalDate.now()`?

(also, we should inject a `Clock` here in order to not have a hard dependency on the system clock 😄)